omod/src/main/java/org/openmrs/module/owa/filter/OwaFilter.java

```diff
@@ -16,30 +15,35 @@
 import javax.servlet.ServletRequest;
 import javax.servlet.ServletResponse;
 import javax.servlet.http.HttpServletRequest;
-
-import org.apache.commons.lang.StringUtils;
-import org.openmrs.api.context.Context;
-import org.openmrs.module.owa.AppManager;
+import java.io.IOException;
 
 /**
  * @author sunbiz
  */
 public class OwaFilter implements Filter {
-	
+
+	public static final String DEFAULT_BASE_URL = "/owa";
+
 	private String openmrsPath;
-	
+
 	@Override
 	public void init(FilterConfig fc) throws ServletException {
 		openmrsPath = fc.getServletContext().getContextPath();
 	}
-	
+
 	@Override
 	public void doFilter(ServletRequest req, ServletResponse res, FilterChain chain) throws IOException, ServletException {
 		HttpServletRequest request = (HttpServletRequest) req;
-		String requestURL = request.getRequestURL().toString();
-		
-		String owaBasePath = Context.getAdministrationService().getGlobalProperty(AppManager.KEY_APP_BASE_URL, "http://localhost:8080/openmrs/owa");
-		
+		String owaBasePath = Context.getAdministrationService().getGlobalProperty(AppManager.KEY_APP_BASE_URL,
+				DEFAULT_BASE_URL);
+
+		String requestURL = null;
+		if (owaBasePath.contains("://")) {

```

Comment:
I wanted to preserve option to set full appBaseUrl, and I thought that schema separator '://' would be good flag for checking if appBaseUrl is full URL, or relative to context path. I've extracted this part as method and commented it. Besides I've added note about it in README.
 
Checking for emptiness would be redundant, owaBasePath is always either GP, or `/owa` if GP is empty.